OSDOCS-241: Initial CLI reference docs for dev commands

bergerhoffer opened the pull request:
@openshift/team-documentation Can someone please peer review? And also look closely to make sure that I didn't miss anything with adding a new heading (CLI Reference) and assembly? Thanks!

Preview: http://file.rdu.redhat.com/~ahoffer/2019.02.19/OSDOCS-241-cli-dev/cli_reference/developer-cli-commands.html

bergerhoffer commented:
@soltysh Can you please tech review this? Thanks!

This is the first of several sections that will be added to the CLI Reference section of the 4.0 docs. This is just the Developer CLI Commands (oc <command>). Another reference section for the Administrator CLI commands (oc adm <command>) will be added eventually, along with other sections.

Preview: http://file.rdu.redhat.com/~ahoffer/2019.02.19/OSDOCS-241-cli-dev/cli_reference/developer-cli-commands.html

soltysh commented on modules/cli-developer-advanced.adoc:
```diff
@@ -0,0 +1,147 @@
+
+.Example: Update the `spec.unschedulable` field for node `node1` to `true`
+----
+$ oc patch node node1 -p '{"spec":{"unschedulable":true}}'
```
It would be nice to use a consistent naming scheme which is `<type>/<name>` so here `node/node1`.

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Add an annotation to a route
+----
+$ oc annotate route test-route haproxy.router.openshift.io/ip_whitelist="192.168.1.10"
```
`route/test-route`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Remove the annotation from the route
+----
+$ oc annotate route test-route haproxy.router.openshift.io/ip_whitelist-
```
`route/test-route`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Autoscale to a minimum of 2 and maximum of 5 Pods
+----
+$ oc autoscale dc/parksmap-katacoda --min=2 --max=5
```
Always use full names in docs, so `deploymentconfig` instead of `dc`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Delete a Pod named `parksmap-katacoda-1-qfqz4`
+----
+$ oc delete pods parksmap-katacoda-1-qfqz4
```
`pod/parksmap-katacoda-1-qfqz4`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Describe a Deployment named `example`
+----
+$ oc describe deployments example
```
`deployment/example`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Edit a deployment configuration using the default editor
+----
+$ oc edit dc/parksmap-katacoda
```
`deploymentconfig`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Edit a deployment configuration using a different editor
+----
+$ OC_EDITOR="nano" oc edit dc/parksmap-katacoda
```
`deploymentconfig`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Edit a deployment configuration in JSON format
+----
+$ oc edit dc/parksmap-katacoda -o json
```
`deploymentconfig`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Expose a service
+----
+$ oc expose service parksmap-katacoda
```
`service/parksmap-katacoda`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Get details about the `python` deployment configuration in JSON format
+----
+$ oc get dc python -o json
```
`deploymentconfig/python`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Update the `python-1-mz2rf` Pod with the label `status` set to `unhealthy`
+----
+$ oc label pods python-1-mz2rf status=unhealthy
```
`pod/python-1-mz2rf`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Scale the `ruby-app` deployment configuration to 3 Pods
+----
+$ oc scale dc ruby-app --replicas=3
```
`deploymentconfig/ruby-app`

soltysh commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+.Example: Sets the name of a secret on a build configuration
+----
+$ oc set build-secret --source bc/mybc mysecret
```
`buildconfig`

soltysh commented on modules/cli-developer-build-deploy.adoc:
```diff
@@ -0,0 +1,100 @@
+
+.Example: Cancel all pending builds from the `python` build configuration
+----
+$ oc cancel-build bc/python --state=pending
```
`buildconfig` instead of `bc`

soltysh commented on modules/cli-developer-build-deploy.adoc:
```diff
@@ -0,0 +1,100 @@
+
+.Example: Roll back to the last successful Deployment
+----
+$ oc rollout undo dc/php
```
`deploymentconfig`

soltysh commented on modules/cli-developer-build-deploy.adoc:
```diff
@@ -0,0 +1,100 @@
+
+.Example: Start a new rollout for a deployment configuration with its latest state
+----
+$ oc rollout latest dc/php
```
`deploymentconfig`

soltysh commented on modules/cli-developer-other.adoc:
```diff
@@ -0,0 +1,47 @@
+[id='cli-other-developer-commands-{context}']
+= Other developer CLI commands
+
+== ex
```
I would skip `oc ex` commands, this are experimental and may change so I'd not document them.

soltysh commented on modules/cli-developer-troubleshooting.adoc:
```diff
@@ -0,0 +1,108 @@
+
+.Example: Debug the `python` Deployment
+----
+$ oc debug dc/python
```
`deploymentconfig`

bergerhoffer commented:
Thanks a lot @soltysh! I've incorporated your suggestions and updated the preview: http://file.rdu.redhat.com/~ahoffer/2019.02.19/OSDOCS-241-cli-dev/cli_reference/developer-cli-commands.html

adellape commented on _topic_map.yml:
```diff
@@ -398,3 +398,10 @@ Topics:
 - Name: Uninstalling the EFK stack
   File: efk-logging-uninstall
+---
+Name: CLI Reference
```
s/Reference/reference/

adellape commented on cli_reference/developer-cli-commands.adoc:
```diff
@@ -0,0 +1,27 @@
+[id='cli-developer-commands']
+= Developer CLI Commands
```
s/Commands/commands/

adellape commented on _topic_map.yml:
```diff
@@ -398,3 +398,10 @@ Topics:
+Dir: cli_reference
+Distros: openshift-*
+Topics:
+- Name: Developer CLI Commands
```
s/Commands/commands/

adellape commented on modules/cli-developer-basic.adoc:
```diff
@@ -0,0 +1,87 @@
+$ oc login
+----
+
+.Example: Log in specifying a username and password
```
s/username/user name/

adellape commented on modules/cli-developer-basic.adoc:
```diff
@@ -0,0 +1,87 @@
+
+.Example: Create a new application from a private remote repository
+----
+$ oc new-app https://github.com/youruser/yourprivaterepo --source-secret=yoursecret
```
Use `<replaceables>`.

adellape commented on modules/cli-developer-basic.adoc:
```diff
@@ -0,0 +1,87 @@
+
+Create a new application by specifying source code, a template, or an image.
+
+.Example: Create a new application from a local git repository
```
Throughout (other than the URLs) : s/git/Git/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+Configure existing application resources.
+
+.Example: Sets the name of a secret on a build configuration
```
Throughout: s/build configuration/BuildConfig/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+Update the annotations on one or more resources.
+
+.Example: Add an annotation to a route
```
Throughout (outside of code blocks): s/route/Route/

adellape commented on modules/cli-developer-advanced.adoc:
```diff
@@ -0,0 +1,147 @@
+
+== idle
+
+Idle scalable resources. An idled service will automatically become unidled when
```
s/service/Service/

adellape commented on modules/cli-developer-advanced.adoc:
```diff
@@ -0,0 +1,147 @@
+Idle scalable resources. An idled service will automatically become unidled when
+it receives traffic or it can be manually unidled using the `oc scale` command.
+
+.Example: Idle the `ruby-app` service
```
s/service/Service/

adellape commented on modules/cli-developer-advanced.adoc:
```diff
@@ -0,0 +1,147 @@
+
+Observe changes to resources and take action on them.
+
+.Example: Observe changes to services
```
s/services/Services/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+== expose
+
+Expose a service externally as a route.
```
s/service/Service/
s/route/Route/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+Expose a service externally as a route.
+
+.Example: Expose a service
```
s/service/Service/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+$ oc expose service/parksmap-katacoda
+----
+
+.Example: Expose a service and specify the host name
```
s/service/Service/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+
+== scale
+
+Set the desired number of replicas for a replication controller or a deployment
```
s/replication controller/ReplicationController/
s/deployment configuration/DeploymentConfig/

adellape commented on modules/cli-developer-application-management.adoc:
```diff
@@ -0,0 +1,170 @@
+Set the desired number of replicas for a replication controller or a deployment
+configuration.
+
+.Example: Scale the `ruby-app` deployment configuration to 3 Pods
```
Since it's not referring directly to a value right here?: s/3/three/

adellape commented on modules/cli-developer-troubleshooting.adoc:
```diff
@@ -0,0 +1,108 @@
+
+== logs
+
+Retrieve the log output for a specific build, build configuration, deployment
```
Throughout: s/build configuration/BuildConfig/
Throughout: s/deployment configuration/DeploymentConfig/

adellape commented on modules/cli-developer-troubleshooting.adoc:
```diff
@@ -0,0 +1,108 @@
+Create and run a particular image. By default, this creates a deployment
+configuration to manage the created containers.
+
+.Example: Start an instance of the `perl` image with 3 replicas
```
Same re: s/3/three/ ?

adellape commented on modules/cli-developer-other.adoc:
```diff
@@ -0,0 +1,38 @@
+
+== plugin
+
+List the available plugins on the user's `PATH`.
```
s/plugins/plug-ins/

adellape commented on modules/cli-developer-other.adoc:
```diff
@@ -0,0 +1,38 @@
+
+List the available plugins on the user's `PATH`.
+
+.Example: List available plugins
```
s/plugins/plug-ins/

bergerhoffer commented:
Thanks for the review @adellape, updated PR and preview.
